### DISCUSSION TITLE
Snapshot count for time interval (last 7 days) for each VM usimg pyvmomi

### DISCUSSION THREAD

**Discussion by sathishkotturi** [vmware/pyvmomi#1010] [General]
Iam trying  to get snapshot  count  for all vm  every  7 days. I need to raise alarms if snapshot  count exceeds the limit.I connected to vcenter using pyvmomi  and unable to fetch the required  details. Can some one help me here?

**Comment by sathishkotturi** [comment]
@tianhao64 @hartsock @DanielDraganov @mstagi can you please help here

**Comment by li-jun-fei** [comment]
`def get_vm_snap_createtime(vm_rootSnapList:list,vm_snap_create_list=[]) ->list:    
    if len(vm_rootSnapList)>0:
        for i in range(len(vm_rootSnapList)):
          vm_snap_create_list.append(vm_rootSnapList[i].createTime.strftime("%m/%d/%Y, %H:%M:%S"))
          if  len(vm_rootSnapList[i].childSnapshotList)>0:      
            get_vm_snap_createtime(vm_rootSnapList[i].childSnapshotList,vm_snap_create_list)
    return vm_snap_create_list  `

get_vm_snap_createtime(vm.snapshot.rootSnapshotList)
this function will return a list of  every snapshot's create time. the length of the returned list is the snapshot's count
